hands-on-exercise/1-native/4-add-message.md

```diff
@@ -0,0 +1,384 @@
+---
+title: "Add your first message"
+order: 5
+description: Where you make it possible to create a game
+tags:
+  - guided-coding
+  - cosmos-sdk
+---
+
+# Add your first message
+
+In the [previous section](./3-add-game.md), you added a game storage space in the checkers module storage. It is now time to actually put games in storage:
+
+* Add a message type to create a game.
+* Add all the necessary keeper and message server functions.
+* Add CLI commands.
+
+## Add a game creation message
+
+With the storage ready to receive games, it is time to add a message to let players create games:
+
+* A game starts with an initial board and turn, as per the rules. The game creator **does not choose** how the initial pieces are placed.
+* A game needs an **unused index** to be created. For simplicity's sake, the index will be passed along with the message.
+
+You are going to:
+
+* Create the message type.
+* Compile Protobuf.
+* Create a message server next to the keeper.
+* Handle the message.
+* Register the necessary elements in the module.
+
+### The game creation object type and Protobuf service
```

Comment:
Is there a difference between "message type" and "object type"? Line 27 uses the first, here the second.

Reply:
There is but here they are used interchangeably...

```suggestion
### The game creation message type and Protobuf service
```